### ISSUE TITLE
Heartbeat is blocked by other scheduled tasks

### ISSUE THREAD

**Issue by matt-richardson** [OctopusDeploy/Issues#4241]
Heartbeat is treated the same as any other scheduled task, when it needs to be reliably running every 5 seconds. If another task takes a long time (ie, a locked file causes a task to retry), it can mean that it doesn't heartbeat in time.
